kindelia_core/src/node.rs

```diff
@@ -774,7 +773,7 @@ impl<C: ProtoComm, S: BlockStorage> Node<C, S> {
     let (query_sender, query_receiver) = mpsc::sync_channel(1);
 
     let genesis_stmts =
-      parser::parse_code(constants::GENESIS_CODE).expect("Genesis code parses");
+      parser::parse_code(&genesis_code(network_id).unwrap()).expect("Genesis code parses");
```

Comment:
This should not implicitly load another file (or even do IO at all) but receive the source code (or the already parsed statements from outside).

`Node` implements a core logic that we are decoupling the most possible from IO, so we can mock on tests, test/benchmark multiple implementations etc. Take a look at `C: ProtoComm` and `S: BlockStorage` for example.

Reply:
yeah that's fine.  I actually started out doing it that way but it was going to require more changes to callers, so I went with the approach that changed fewest LOC.

So the basic idea then, as I see it, is that `kindelia start` command receives the network_id and the cli layer reads the corresponding `~/kindelia/genesis/<network-id>.kdl file`, parses the statements, and then passes `Vec<Statement>` to `Node::new()`.   Sound good?

Reply:
Node::new now receives &[Statement], so marking this resolved.